File: uri/uri_python/estruturas_e_bibliotecas/p1063.py
N = int(input())
while N != 0:
    e1 = input().split(' ')
    entry1 = []
    for i in range(len(e1)):
        if len(e1[i]) > 0:
            if ord(e1[i]) >= 97 and ord(e1[i]) <= 122:
                entry1.append(e1[i])
    e2 = input().split(' ')
    entry2 = []
    for i in range(len(e2)):
        if len(e2[i]) > 0:
            if ord(e2[i]) >= 97 and ord(e2[i]) <= 122:
                entry2.append(e2[i])
    stack = ['-', entry1.pop(0)]
    out = 'I'
    while True:
        #print('s: ', stack)
        #print('e2: ', entry2)
        #print('e1: ', entry1)
        if stack[len(stack)-1] == entry2[0]:
            out += 'R'
            stack.pop()
            entry2.pop(0)
        else:
            if len(entry1) == 0:
                break
            stack.append(entry1.pop(0))
            out += 'I'
        if len(entry2) == 0:
            break
        
    if len(entry2) == 0:
        print(out)
    else:
        print(out, "Impossible")
    N = int(input())

File: uri/uri_cpp/estruturas_e_bibliotecas/p1063.cpp
#include <iostream>
#include <sstream>
#include <string>
#include <stdlib.h>
#include <stdio.h>
#include <vector>

using namespace std;

int main() {
    int N, i;
    char e1, e2;
    cin >> N;

    while (N != 0) {
        vector <char> entry1;
        entry1.push_back('-');
        for (i = 0; i < N; i++) {
            scanf(" %c", &e1);
            if (e1 >= 97 and e1 <= 122)
                entry1.push_back(e1);
        }
        vector <char> entry2;
        entry2.push_back('-');
        for (i = 0; i < N; i++) {
            scanf(" %c", &e2);
            if (e2 >= 97 and e2 <= 122)
                entry2.push_back(e2);
        }
        vector <char> stack;
        stack.push_back('-');
        stack.push_back(entry1[1]);
        entry1.erase(entry1.begin()+1);
        string out = "I";
        while (true) {
            if (stack[stack.size()-1] == entry2[1]) {
                out += 'R';
                stack.pop_back();
                entry2.erase(entry2.begin()+1);
            } else {
                if (entry1.size() == 1)
                    break;
                stack.push_back(entry1[1]);
                entry1.erase(entry1.begin()+1);
                out += 'I';
            }
            if (entry2.size() == 1)
                break;
        }
        if (entry2.size() == 1) {
            cout <<out << endl;
        } else {
            cout << out << " Impossible" << endl;
        }
        cin >> N;
    }
    return 0;
}